src/main/groovy/com/wooga/gradle/io/ProcessExecutor.groovy

```diff
@@ -0,0 +1,229 @@
+    Closure closure
+    Boolean _ignoreExitValue = false
+    String workingDirectory
+    OutputStream _standardOutput, _standardError
+
+    //------------------------------------------------------------------------/
+    // Constructors
+    //------------------------------------------------------------------------/
+    private ProcessExecutor(ProcessOperations processOperations) {
+        function = { a -> processOperations.exec(a as Action<? super org.gradle.process.ExecSpec>) }
+    }
+
+    private ProcessExecutor(Project project) {
+        function = { a -> project.exec(a as Action<? super org.gradle.process.ExecSpec>) }
+    }
+
+    /**
+     * Constructs the executor from an object implementing the {@code ProcessOperations} interface
+     */
+    static ProcessExecutor from(ProcessOperations processOperations) {
+        new ProcessExecutor(processOperations)
+    }
+
+    /**
+     * Constructs the executor from a Gradle {@code Project}
+     */
+    static ProcessExecutor from(Project project) {
+        new ProcessExecutor(project)
+    }
+
+    /**
+     * Constructs the executor from an {@code ExecSpec}
+     */
+    static ProcessExecutor from(ExecSpec spec) {
+        new ProcessExecutor(spec.processOperations)
+            .withExecutable(spec.executablePath.getOrNull())
+            .withWorkingDirectory(spec.workingDirectory.getOrNull())
+    }
+
+    /**
+     * Constructs the executor from an {@code Executable}
+     */
+    static ProcessExecutor fromExecutable(Executable executable) {
+        from(executable).withArguments(executable, true)
+    }
+
+    //------------------------------------------------------------------------/
+    // Configuration
+    //------------------------------------------------------------------------/
+    ProcessExecutor withExecutable(String filePath) {
+        executablePath = filePath
+        this
+    }
+
+    ProcessExecutor withExecutable(File file) {
+        withExecutable(file.path)
+    }
+
+    ProcessExecutor withArguments(ArgumentsSpec spec, Boolean includeEnvironment = true) {
+        withArguments(spec.arguments.getOrNull())
+        if (includeEnvironment) {
+            withEnvironment(spec.environment.getOrNull())
+        }
+        this
+    }
+
+    ProcessExecutor withArguments(Iterable<String> args) {
+        this.arguments = args.toList()
+        this
+    }
+
+    ProcessExecutor withArguments(String arg) {
+        this.arguments = [arg]
+        this
+    }
+
+    /**
+     * Sets the working directory for the execution
+     */
+    ProcessExecutor withWorkingDirectory(String value) {
+        this.workingDirectory = value
+        this
+    }
+
+    ProcessExecutor withEnvironment(Map<String, ?> environment, Boolean append = true) {
+        if (environment != null) {
+            this.environment = environment
+            this.appendToEnvironment = append
+        }
+        this
+    }
+
+    ProcessExecutor withIgnoredExitValue() {
```

Comment:
```suggestion
    ProcessExecutor withIgnoreExitValue() {
```

Reply:
What about the newer name? I would like to name it `ignoreExitValue()`